developer-docs-site/docs/guides/nfts/mint-nft-v2-cli.md

```diff
@@ -0,0 +1,918 @@
+                  string:"https://www.link-to-your-collection-image.com"  \
+                  u64:3                                                   \
+                  u64:5                                                   \
+                  u64:100                                                 \
+                  string:"Krazy Kangaroo #1"                              \
+                  string:"https://www.link-to-your-token-image.com"       
+```
+
+Next we need to get the resource address for the contract with our view function.
+
+```shell
+aptos move view --function-id default::create_nft_with_resource_account::get_resource_address \
+                --profile default \
+                --args string:"Krazy Kangaroos"
+```
+
+Now we call this function as a user, which we simulate with our `nft-receiver` profile:
+
+```shell
+aptos move run --function-id default::create_nft_with_resource_account::mint \
+               --profile nft-receiver \
+               --args address:YOUR_RESOURCE_ADDRESS_HERE
+```
+
+Great! Now you've created the collection as an owner and requested to mint as a user and received the newly minted NFT.
+
+It may not feel different since you're acting as the owner and the receiver all from the command line, but in an actual dapp this user flow makes much more sense than before.
+
+In the first section, the user has to wait for the owner of the contract to mint and send them an NFT. In the second section, the user can request to mint and receive an NFT themselves.
+
+## 3. Adding restrictions: a whitelist, an end time, an admin, and an enabled flag
+
+We're still missing some very common features for NFT minting contracts:
+
+1. A whitelist that restricts minting to whitelisted addresses
+2. The ability to add/remove addresses from the whitelist
+3. A start* and end time
+4. The ability to enable or disable the mint
+5. An admin model: restricting using these functions to an assigned admin account
+
+*We add the start time in part 4 to keep this section brief.
+
+### Adding the new configuration options
+
+We need to add the expiration timestamp, the enabled flag, and the admin address to the mint configuration resource:
+
+```rust
+struct MintConfiguration has key {
+    // ...
+    whitelist: Table<address, bool>,
+    expiration_timestamp: u64,
+    minting_enabled: bool,
+    admin: address,
+}
+```
+
+Note that we're storing a `bool` in the whitelist as the value in each key: value pair. We won't use it in this tutorial, but you could easily use it to limit each account to 1 mint or even use an integer type to limit it to an arbitrary number of mints.  
+
+When we initialize the collection, we create a default empty whitelist, an expiration timestamp that's one second in the past, and disable the mint:
+
+```rust
+public entry fun initialize_collection( /* ... */ ) {
+    // ...
+
+    move_to(&resource_signer, MintConfiguration {
+        // ...
+        whitelist: table::new<address, bool>(),
```

Comment:
Let's use smart_table here instead

Reply:
I use it in the next section! Is it confusing if I go from `table` to `smart_table` or you think I should swap it still